src/NServiceBus.AcceptanceTests/Recoverability/Retries/When_performing_slr_with_custom_policy.cs

```diff
@@ -1,34 +1,39 @@
 ﻿namespace NServiceBus.AcceptanceTests.Recoverability.Retries
 {
     using System;
+    using System.Collections.Generic;
+    using System.Linq;
     using System.Threading.Tasks;
     using AcceptanceTesting;
     using EndpointTemplates;
     using Features;
     using NUnit.Framework;
-    using Transports;
 
     public class When_performing_slr_with_custom_policy : NServiceBusAcceptanceTest
     {
         [Test]
-        public async Task Should_expose_headers_to_policy()
+        public async Task Should_expose_error_context_to_policy()
         {
             var context = await Scenario.Define<Context>()
-                .WithEndpoint<Endpoint>(b => 
+                .WithEndpoint<Endpoint>(b =>
                     b.When(bus => bus.SendLocal(new MessageToBeRetried()))
                      .DoNotFailOnErrorMessages())
-                .Done(c => c.MessageMovedToErrorQueue)
+                .Done(c => c.FailedMessages.Any())
                 .Run();
 
-            Assert.That(context.FLRetriesHeaderAvailable, Is.True, "Could not find FLRetries header");
-            Assert.That(context.RetriesHeaderAvailable, Is.True, "Could not find Retries header");
+            Assert.That(context.HeaderValues.Count, Is.EqualTo(2), "because the custom policy should have been invoked twice");
+            Assert.That(context.HeaderValues[0].Message, Is.Not.Null);
+            Assert.That(context.HeaderValues[0].Exception, Is.TypeOf<SimulatedException>());
+            Assert.That(context.HeaderValues[0].SecondLevelRetryAttempt, Is.EqualTo(1));
+            Assert.That(context.HeaderValues[1].Message, Is.Not.Null);
+            Assert.That(context.HeaderValues[1].Exception, Is.TypeOf<SimulatedException>());
+            Assert.That(context.HeaderValues[1].SecondLevelRetryAttempt, Is.EqualTo(2));
         }
 
         class Context : ScenarioContext
         {
-            public bool FLRetriesHeaderAvailable { get; set; }
             public bool MessageMovedToErrorQueue { get; set; }
-            public bool RetriesHeaderAvailable { get; set; }
+            public List<SecondLevelRetryContext> HeaderValues { get; } = new List<SecondLevelRetryContext>();
```

Comment:
Is the name of the property still accurate? Since the name `SecondLevelRetryContext` doesn't imply header values